### PR TITLE
Add support for ZMON metrics in autoscaler config

### PR DESCRIPTION
**Pull request by mikkeloscar**
Extends the custom `autoscaler` field to also support the metric type `ZMON`. Without this users have to switch to the legacy `horizontalPodAutoscaler` configuration in case they want to depend on ZMON for scaling.

https://github.com/zalando-incubator/kube-metrics-adapter#zmon-collector

### CODE REVIEW

File: pkg/apis/zalando.org/v1/types.go [hunk 1]
```diff
@@ -170,6 +170,38 @@ type MetricsQueue struct {
...
+	CountZMONMetricAggregator ZMONMetricAggregatorType = "count"
+	FirstZMONMetricAggregator ZMONMetricAggregatorType = "first"
+	LastZMONMetricAggregator  ZMONMetricAggregatorType = "last"
+	MaxZMONMetricAggregator   ZMONMetricAggregatorType = "max"
+	MinZMONMetricAggregator   ZMONMetricAggregatorType = "min"
+	SumZMONMetricAggregator   ZMONMetricAggregatorType = "sum"
+	DiffZMONMetricAggregator  ZMONMetricAggregatorType = "diff"
+)
+
+// MetricsZMON specifies the ZMON check whcih should be used for scaling.
```
Comment (jonathanbeber) [inline pkg/apis/zalando.org/v1/types.go:190]: nit: `s/whcih/which/g`

File: docs/stack_crd.yaml [hunk 2]
```diff
@@ -190,6 +190,40 @@ spec:
                           - Ingress
                           - ZMON
                           type: string
+                        zmon:
+                          description: MetricsZMON specifies the ZMON check whcih should be used for scaling.
```
Comment (oporkka) [inline docs/stack_crd.yaml:194]:
```suggestion
                          description: MetricsZMON specifies the ZMON check which should be used for scaling.
```

File: docs/stackset_crd.yaml [hunk 3]
```diff
@@ -363,6 +363,40 @@ spec:
                                   - Ingress
                                   - ZMON
                                   type: string
+                                zmon:
+                                  description: MetricsZMON specifies the ZMON check whcih should be used for scaling.
```
Comment (oporkka) [inline docs/stackset_crd.yaml:367]:
```suggestion
                                  description: MetricsZMON specifies the ZMON check which should be used for scaling.
```

**Comment by jonathanbeber** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**